restli-client/src/main/java/com/linkedin/restli/client/base/EntityActionRequestBuilderBase.java

```diff
@@ -0,0 +1,27 @@
+package com.linkedin.restli.client.base;
+
+import com.linkedin.restli.client.ActionRequest;
+import com.linkedin.restli.client.RestliRequestOptions;
+import com.linkedin.restli.common.ResourceSpec;
+
+/**
+ * The abstract base class for all generated request builders classes for entity-level actions.
+ */
+public abstract class EntityActionRequestBuilderBase<K, V, RB extends ActionRequestBuilderBase<K, V, RB>>
+        extends ActionRequestBuilderBase<K, V, RB>
+{
+    protected EntityActionRequestBuilderBase(String baseUriTemplate, Class<V> elementClass, ResourceSpec resourceSpec,
+        RestliRequestOptions requestOptions)
+    {
+      super(baseUriTemplate, elementClass, resourceSpec, requestOptions);
+    }
+
+    @Override
+    public ActionRequest<V> build() {
+      if (!hasId())
```

Comment:
what about entity level resource identified in the simple resource. It is a legit case but with no entity key. in this case resourcelevel.entity == resourcelevel.any;

Will this request builder also fail in this case?

Reply:
Good call, let me look into how the simple resource case works.

Reply:
I've just added a commit to this PR that shows that entity level actions in simple resources are still working with my change. I added integration tests for actions with both of the supported resource levels ANY and ENTITY.

The reason why this isn't broken by my other changes is kind of interesting. For simple resources, entity level actions in the restspec.json do not appear under the "entity" JSON key. They appear as top level actions similar to where collection level actions appear in a restspec.json for a collection resource. So from a code gen perspective they're not really entity level actions.

Reply:
One last request on this:

Can you add what you explain as a comment/documentation to the code?

Reply:
I've added to the class-level javadoc and pushed that commit. Take a look and let me know what you think.